dxp/sennajs_integration.md

```diff
@@ -0,0 +1,117 @@
+# Senna.js integration in DXP
+
+This article describes how [Senna.js](https://github.com/liferay/senna.js) is integrated in DXP and how it works under the hood.
+
+### Introduction
+
+The infrastructure around Senna.js is located in the [`frontend-js-spa-web`](https://github.com/liferay/liferay-portal/tree/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web) module.
+
+In this module, a Senna.js [`App`](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/app/App.js) instance is created, with two routes which are handled by a specific [`Screen`](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/screen/Screen.js) instance.
+
+-   The [`ActionURLScreen`](https://github.com/liferay/liferay-portal/blob/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/ActionURLScreen.es.js) class is used for all requests made to "action" URLs, which are used to post forms. It makes sure that the http method used is `POST` and checks that the `p_p_lifecycle` URL parameter has the correct value when submitting a form.
```

Comment:
> has the correct value

Should we mention what that value is? Off the top of my head, I don't know.

Reply:
Yes that's actually a good idea, the value should be `0` - or at least that's what I understand when I read [this](https://github.com/liferay/liferay-portal/blob/b619978445e3be09526803b27447b5ddab95ca6f/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/ActionURLScreen.es.js#L53-L55)